src/read.c

```diff
@@ -4670,25 +4688,29 @@ static avifResult avifMetaFindAlphaItem(avifMeta * meta,
                         // One of the following invalid cases:
                         // * Multiple items are claiming to be the alpha auxiliary of the current item.
                         // * Alpha auxiliary is dimg for another item.
-                        avifFree(alphaItemIndices);
+                        avifFree(dimgIdxToAlphaItemIdx);
                         *alphaItem = NULL;
                         *isAlphaItemInInput = AVIF_FALSE;
                         return AVIF_RESULT_INVALID_IMAGE_GRID;
                     }
-                    alphaItemIndices[alphaItemCount++] = j;
+                    if (item->dimgIdx >= tileCount || dimgIdxToAlphaItemIdx[item->dimgIdx] != itemIndexNotSet) {
+                        avifFree(dimgIdxToAlphaItemIdx);
+                        *isAlphaItemInInput = AVIF_FALSE;
```

Comment:
Should we add `*alphaItem = NULL;` here? See line 4669 and line 4692.

Reply:
It seems there is no impact of doing so in case of error.

Reply:
You are right. This function usually sets `*alphaItem` to `NULL` and `*isAlphaItemInInput` to `AVIF_FALSE` when it returns an error, but it isn't consistent. The callers of this function ignore the output parameters when the function fails. So this function can actually follow the common policy if not touching the output parameters when it fails. We can write a CL to do that later.

For now, let's follow the current policy of setting the output parameters to `NULL` and `AVIF_FALSE` before returning an error.

Reply:
Done in https://github.com/AOMediaCodec/libavif/pull/2330.